### PR TITLE
[Docs] cts documentation

### PR DESCRIPTION
**Pull request by luarss**
Changes
- Missing parameters 
- Formatting changes
- Removal of this section
image

https://openroad--3698.org.readthedocs.build/en/3698/main/src/cts/README.html

### CODE REVIEW

File: src/cts/README.md [hunk 1]
```diff
@@ -1,143 +1,138 @@
...
-                     [-clk_nets <list_of_clk_nets>] \
-                     [-distance_between_buffers] \
-                     [-branching_point_buffers_distance] \
-                     [-clustering_exponent] \
-                     [-clustering_unbalance_ratio] \
-                     [-sink_clustering_enable] \
-                     [-sink_clustering_size <cluster_size>] \
-                     [-sink_clustering_max_diameter <max_diameter>]
+```tcl
+clock_tree_synthesis
```
Comment (vvbandeira) [inline src/cts/README.md:39]: Missing:
```
[-num_static_layers] \
[-sink_clustering_buffer] \
```

File: src/cts/README.md [hunk 2]
```diff
@@ -1,143 +1,138 @@
...
-    this parameter is omitted, TritonCTS looks for the clock roots automatically.
-
+#### Options
+
+| Switch Name | Description |
+| ----- | ----- |
+| `-buf_list` | Tcl list of master cells (buffers) that will be considered when making the wire segments (e.g. `{BUFXX, BUFYY}`). |
+| `-root_buffer` | The master cell of the buffer that serves as root for the clock tree. If this parameter is omitted, the first master cell from `-buf_list` is taken. |
+| `-wire_unit` | Minimum unit distance between buffers for a specific wire. If this parameter is omitted, the code gets the value from ten times the height of `-root_buffer`. |
+| `-clk_nets` | String containing the names of the clock roots. If this parameter is omitted, TritonCTS looks for the clock roots automatically. |
```
Comment (vvbandeira) [inline src/cts/README.md:60]:
```suggestion
| `-clk_nets` | String containing the names of the clock roots. If this parameter is omitted, `cts` automatically looks for the clock roots. |
```

File: src/cts/README.md [hunk 3]
```diff
@@ -1,143 +1,138 @@
...
+| `-distance_between_buffers` | Distance (in microns) between buffers that TritonCTS should use when creating the tree. When using this parameter, the clock tree algorithm is simplified, and only uses a fraction of the segments from the LUT. |
```
Comment (vvbandeira) [inline src/cts/README.md:61]:
```suggestion
| `-distance_between_buffers` | Distance (in microns) between buffers that `cts` should use when creating the tree. When using this parameter, the clock tree algorithm is simplified and only uses a fraction of the segments from the LUT. |
```

File: src/cts/README.md [hunk 4]
```diff
@@ -1,143 +1,138 @@
...
+| `-branching_point_buffers_distance` | Distance (in microns) that a branch has to have in order for a buffer to be inserted on a branch end-point. This requires the `-distance_between_buffers` value to be set. |
+| `-clustering_exponent` | Value that determines the power used on the difference between sink and means on the CKMeans clustering algorithm. (default 4, integer). |
+| `-clustering_unbalance_ratio` | Value that determines the maximum capacity of each cluster during CKMeans. A value of 50% means that each cluster will have exactly half of all sinks for a specific region (half for each branch). (default 0.6, float). |
```
Comment (vvbandeira) [inline src/cts/README.md:64]:
```suggestion
| `-clustering_unbalance_ratio` | Value determines each cluster's maximum capacity during CKMeans. A value of `0.5` (i.e., 50%) means that each cluster will have exactly half of all sinks for a specific region (half for each branch). (default 0.6, float). |
```

File: src/cts/README.md [hunk 5]
```diff
@@ -1,143 +1,138 @@
...
+| `-sink_clustering_enable` | Enables pre-clustering of sinks to create one level of sub-tree before building H-tree. Each cluster is driven by buffer which becomes end point of H-tree structure. |
+| `-sink_clustering_size` | Specifies the maximum number of sinks per cluster. (default 20, integer). |
+| `sink_clustering_max_diameter` | Specifies maximum diameter (in microns) of sink cluster (default 50, integer). |
+| `-clk_nets` | String containing the names of the clock roots. If this parameter is omitted, TritonCTS looks for the clock roots automatically. |
 
 ### Report CTS
 
 Another command available from TritonCTS is `report_cts`. It is used to
-extract metrics after a successful `clock_tree_synthesis` run. These
```
Comment (vvbandeira) [inline src/cts/README.md:94]: As discussed, the same of the original work should be mentioned in the first paragraph of the readme. In all other places we should refer to the OR module name, in this case `cts`.

File: src/cts/README.md [hunk 6]
```diff
@@ -1,143 +1,138 @@
...
 
-Argument description:
+#### Options
 
--   `-out_file` (optional) is the file containing the TritonCTS reports.
-    If this parameter is omitted, the metrics are shown on the standard
-    output.
+| Switch Name | Description |
+| ----- | ----- |
+| `-out_file` | The file containing the TritonCTS reports. If this parameter is omitted, the metrics are streamed to `stdout` and not saved. |
```
Comment (vvbandeira) [inline src/cts/README.md:89]:
```suggestion
| `-out_file` | The file to save `cts` reports. If this parameter is omitted, the report is streamed to `stdout` and not saved. |
```

**Review by vvbandeira — CHANGES_REQUESTED** [review]

File: src/cts/README.md (hunk 1 above)
Comment (luarss) [reply src/cts/README.md:39]: I need some clarification on this, as I am not sure on the purpose on these switches.

Can you help me ping the tool maintainer?

Comment (vvbandeira) [reply src/cts/README.md:40]: He is on vacation this week; we can keep a generic description and improve it later. This is good enough for now, thanks.

File: src/cts/README.md [hunk 7]
```diff
@@ -1,143 +1,143 @@
...
--   `-clk_nets` is a string containing the names of the clock roots. If
-    this parameter is omitted, TritonCTS looks for the clock roots automatically.
+#### Options
 
+| Switch Name | Description |
+| ----- | ----- |
+| `-out_file` | The file to save `cts` reports. If this parameter is omitted, the report is streamed to `stdout` and not saved. |
 
-### Report CTS
+### Useful developer functions
```
Comment (vvbandeira) [inline src/cts/README.md:96]:
```suggestion
### Useful Developer Commands
```

File: src/cts/README.md [hunk 8]
```diff
@@ -1,143 +1,143 @@
...
 clock_tree_synthesis -root_buf "BUF_X4" \
                      -buf_list "BUF_X4" \
                      -wire_unit 20
-
 report_cts "file.txt"
 ```
 
 ## Regression tests
 
+There are a set of regression tests in `/test`. For more information, refer to this [section](../../README.md#regression-tests).
```
Comment (vvbandeira) [inline src/cts/README.md:115]:
```suggestion
There are a set of regression tests in `./test`. For more information, refer to this [section](../../README.md#regression-tests).
```

**Review by vvbandeira — CHANGES_REQUESTED** [review]

File: src/cts/README.md [hunk 9]
```diff
@@ -1,143 +1,143 @@
...
-`configure_cts_characterization` command.  Use `set_wire_rc` command to
+# Clock Tree Synthesis
+
+The clock tree synthesis module in OpenROAD (`cts`) is based on TritonCTS 
+2.0. It is available from the `clock_tree_synthesis` command.  
+TritonCTS 2.0 performs on-the-fly characterization. Thus there is
+no need to generate characterization data. On-the-fly characterization feature
+can be optionally controlled by parameters specified to 
+`configure_cts_characterization` command. Use `set_wire_rc` command to 
 set clock routing layer.
```
Comment (vvbandeira) [inline src/cts/README.md:9]:
```suggestion
The clock tree synthesis module in OpenROAD (`cts`) is based on TritonCTS
2.0. It is available from the `clock_tree_synthesis` command. TritonCTS 2.0
performs on-the-fly characterization. Thus, there is no need to generate
characterization data. The on-the-fly characterization feature can be optionally
controlled by parameters specified by the `configure_cts_characterization`
command. Use `set_wire_rc` command to set the clock routing layer.
```

**Review by vvbandeira — CHANGES_REQUESTED** [review]

**Review by vvbandeira — APPROVED** [review]

**Pull request merged**